### PR TITLE
RWA-211: Am-role-assignment PACT test

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-211

### Change description ###

Rwa 211: Added am-role-assignment pact test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -55,4 +58,11 @@ withPipeline(type, product, component) {
         steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/site/serenity/**/*'
         steps.archiveArtifacts allowEmptyArchive: true, artifacts: '**/reports/tests/functional/**/*'
     }
+
+    //Temporary enable pact test at PR level
```
Comment (iguannago) [inline Jenkinsfile_CNP:62]: why only temporarily? I mean, it would be even better if you can catch issues on the PR, wouldn't it?

Comment (alexbaeza) [reply Jenkinsfile_CNP:62]: That's true I'll leave on for all Prs then

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/roleassignment/AmRoleAssignmentServiceConsumerTestForCreateRoleAssignment.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                  + "      \"actorIdType\": \"IDAM\",\n"
+                  + "      \"classification\": \"PUBLIC\",\n"
+                  + "      \"grantType\": \"STANDARD\",\n"
+                  + "      \"readOnly\": false,\n"
+                  + "      \"roleCategory\": \"LEGAL_OPERATIONS\",\n"
+                  + "      \"roleName\": \"tribunal-caseworker\",\n"
+                  + "      \"roleType\": \"ORGANISATION\",\n"
+                  + "      \"attributes\": {\n"
+                  + "        \"jurisdiction\": \"IA\",\n"
+                  + "        \"primaryLocation\": \"765324\"\n"
```
Comment (iguannago) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/roleassignment/AmRoleAssignmentServiceConsumerTestForCreateRoleAssignment.java:63]: shall we put the baseLocation here too?

**Review by iguannago — DISMISSED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/roleassignment/AmRoleAssignmentServiceConsumerTestForCreateRoleAssignment.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import static io.pactfoundation.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+public class AmRoleAssignmentServiceConsumerTestForCreateRoleAssignment extends SpringBootContractBaseTest {
+
+    private static final String RAS_CREATE_ROLE_ASSIGNMENT_URL = "/am/role-assignments";
+
+    @Pact(provider = "am_role_assignment_service_create", consumer = "am_org_role_mapping")
```
Comment (ConorH994) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/roleassignment/AmRoleAssignmentServiceConsumerTestForCreateRoleAssignment.java:31]: please change the consumer name as it is over writing our pact in the remote broker

Comment (alexbaeza) [reply src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/roleassignment/AmRoleAssignmentServiceConsumerTestForCreateRoleAssignment.java]: Hi @ConorH994  sorry about that done now

**Review by indvamshi — APPROVED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/roleassignment/AmRoleAssignmentServiceConsumerTestForGetActorById.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    private Map<String, String> getResponseHeaders() {
+        Map<String, String> responseHeaders = Maps.newHashMap();
+        responseHeaders.put("Content-Type",
+            "application/vnd.uk.gov.hmcts.role-assignment-service.get-assignments+json;charset=UTF-8;version=1.0");
+        return responseHeaders;
+    }
+
+}
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/roleassignment/AmRoleAssignmentServiceConsumerTestForGetActorById.java:94]: Do we need a PACT for ->
@DeleteMapping(
        value = "/am/role-assignments/{role-assignment-id}",
        consumes = MediaType.APPLICATION_JSON_VALUE
    )

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**